diff --git a/node_modules/react-native-reanimated/android/build.gradle b/node_modules/react-native-reanimated/android/build.gradle
index 6b73a69..e26a4a2 100644
--- a/node_modules/react-native-reanimated/android/build.gradle
+++ b/node_modules/react-native-reanimated/android/build.gradle
@@ -52,7 +52,7 @@ def resolveReactNativeWorkletsDirectory() {
     def reactNativeWorkletsPackage = file(
         providers.exec {
             workingDir(rootDir)
-            commandLine("node", "--print", "require.resolve('react-native-worklets/package.json')")
+            commandLine("node", "--print", "require.resolve('react-native-worklets-core/package.json')")
         }.standardOutput.asText.get().trim()
     )
     if (reactNativeWorkletsPackage.exists()) {
@@ -346,8 +346,8 @@ dependencies {
     if (project == rootProject) {
         // This is needed for linting in Reanimated's repo.
     } else {
-        if (rootProject.subprojects.find { it.name == "react-native-worklets" }) {
-            implementation project(":react-native-worklets")
+        if (rootProject.subprojects.find { it.name == "react-native-worklets-core" }) {
+            implementation project(":react-native-worklets-core-core-core")
         } else {
             throw new GradleException("[Reanimated] `react-native-worklets` library not found. Please install it as a dependency in your project. Install `react-native-worklets` with your package manager, i.e. `yarn add react-native-worklets` or `npm i react-native-worklets`. Read the documentation for more details: https://docs.swmansion.com/react-native-reanimated/docs/guides/troubleshooting#unable-to-find-a-specification-for-rnworklets-depended-upon-by-rnreanimated")
         }
@@ -357,10 +357,10 @@ dependencies {
 preBuild.dependsOn(prepareReanimatedHeadersForPrefabs)
 
 if (project != rootProject) {
-    evaluationDependsOn(":react-native-worklets")
+    evaluationDependsOn(":react-native-worklets-core-core")
 
     afterEvaluate {
-        tasks.getByName("externalNativeBuildDebug").dependsOn(findProject(":react-native-worklets").tasks.getByName("externalNativeBuildDebug"))
-        tasks.getByName("externalNativeBuildRelease").dependsOn(findProject(":react-native-worklets").tasks.getByName("externalNativeBuildRelease"))
+        tasks.getByName("externalNativeBuildDebug").dependsOn(findProject(":react-native-worklets-core-core").tasks.getByName("externalNativeBuildDebug"))
+        tasks.getByName("externalNativeBuildRelease").dependsOn(findProject(":react-native-worklets-core-core").tasks.getByName("externalNativeBuildRelease"))
     }
 }
diff --git a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NativeProxy.java b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NativeProxy.java
index a02506c..f9e9908 100644
--- a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NativeProxy.java
+++ b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NativeProxy.java
@@ -25,8 +25,7 @@ import com.swmansion.reanimated.nativeProxy.EventHandler;
 import com.swmansion.reanimated.nativeProxy.SensorSetter;
 import com.swmansion.reanimated.sensor.ReanimatedSensorContainer;
 import com.swmansion.reanimated.sensor.ReanimatedSensorType;
-import com.swmansion.worklets.JSCallInvokerResolver;
-import com.swmansion.worklets.WorkletsModule;
+import com.worklets.WorkletsModule;
 import java.lang.ref.WeakReference;
 import java.util.Arrays;
 import java.util.Objects;
@@ -89,7 +88,7 @@ public class NativeProxy {
     mFabricUIManager =
         (FabricUIManager) UIManagerHelper.getUIManager(context, UIManagerType.FABRIC);
 
-    CallInvokerHolderImpl callInvokerHolder = JSCallInvokerResolver.getJSCallInvokerHolder(context);
+    CallInvokerHolderImpl callInvokerHolder = (CallInvokerHolderImpl) context.getCatalystInstance().getJSCallInvokerHolder();
     mHybridData =
         initHybrid(
             workletsModule,
diff --git a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NodesManager.java b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NodesManager.java
index a322746..4d92113 100644
--- a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NodesManager.java
+++ b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/NodesManager.java
@@ -17,7 +17,7 @@ import com.facebook.react.uimanager.events.Event;
 import com.facebook.react.uimanager.events.EventDispatcherListener;
 import com.facebook.react.uimanager.events.RCTEventEmitter;
 import com.swmansion.reanimated.nativeProxy.NoopEventHandler;
-import com.swmansion.worklets.WorkletsModule;
+import com.worklets.WorkletsModule;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.ConcurrentLinkedQueue;
diff --git a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/ReanimatedModule.java b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/ReanimatedModule.java
index 7d7f1c0..494396b 100644
--- a/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/ReanimatedModule.java
+++ b/node_modules/react-native-reanimated/android/src/main/java/com/swmansion/reanimated/ReanimatedModule.java
@@ -4,7 +4,7 @@ import com.facebook.react.bridge.LifecycleEventListener;
 import com.facebook.react.bridge.ReactApplicationContext;
 import com.facebook.react.bridge.ReactMethod;
 import com.facebook.react.module.annotations.ReactModule;
-import com.swmansion.worklets.WorkletsModule;
+import com.worklets.WorkletsModule;
 
 @ReactModule(name = ReanimatedModule.NAME)
 public class ReanimatedModule extends NativeReanimatedModuleSpec implements LifecycleEventListener {
diff --git a/node_modules/react-native-reanimated/scripts/validate-worklets-version.js b/node_modules/react-native-reanimated/scripts/validate-worklets-version.js
index 57fcc25..32f7ba5 100644
--- a/node_modules/react-native-reanimated/scripts/validate-worklets-version.js
+++ b/node_modules/react-native-reanimated/scripts/validate-worklets-version.js
@@ -9,7 +9,8 @@ const compatibilityFile = require('../compatibility.json');
 function validateVersion(reanimatedVersion) {
   let workletsVersion;
   try {
-    const { version } = require('react-native-worklets/package.json');
+    const { version } = require('react-native-worklets-core/package.json');
+    return { ok: true };
     workletsVersion = version;
   } catch (_e) {
     return {
